src/main/java/org/icgc/argo/workflow_management/service/WebLogEventSender.java

```diff
@@ -79,6 +83,22 @@ public void sendFailedPodEvent(String podName) {
     httpClient.sendHttpMessage(endpoint, toJsonString(message));
   }
 
+  public void sendManagementEvent(RunParams params, Event event) {
+    String time =
+        Bolts.format(new Date(), Const.ISO_8601_DATETIME_FORMAT, TimeZone.getTimeZone("UTC"));
+
+    val queueMessage =
+        WorkflowManagementEvent.builder()
+            .event(event.name())
+            .runName(params.getRunName())
+            .runParams(params)
+            .utcTime(time)
+            .build();
+
+    log.info("Sending management event to weblog: Event {} runName {}", event, params.getRunName());
+    httpClient.sendHttpMessage(endpoint, toJsonString(queueMessage));
```

Comment:
Since we are already doing some refactoring. Here are my 2 cents. 

* We shouldn't be relying on the SimpleHttpClient because it comes from the Nextflow dependency. Ideally as much of the service as possible should be independent of Nextflow. 
* If we pick an http client, we should use one that either returns the success/failure of the call or through a callback. That way if errors occur between management and weblog we are not in the dark.

Reply:
Given that this is a void method. We should try to make it non-blocking. Maybe using a non-blocking http client.